beluga/include/beluga/motion/ackerman_drive_model.hpp

```diff
@@ -0,0 +1,291 @@
+// limitations under the License.
+
+#ifndef BELUGA_MOTION_ACKERMAN_DRIVE_MODEL_HPP
+#define BELUGA_MOTION_ACKERMAN_DRIVE_MODEL_HPP
+
+#include <chrono>
+#include <random>
+#include <sophus/se3.hpp>
+#include <tuple>
+
+#include <beluga/type_traits/tuple_traits.hpp>
+#include <beluga/utility/time_stamped.hpp>
+
+#include <beluga/3d_embedding.hpp>
+#include <sophus/se2.hpp>
+#include <sophus/so2.hpp>
+#include <type_traits>
+
+/**
+ * \file
+ * \brief Implementation of a velocity motion model.
+ */
+
+namespace beluga {
+
+/// Velocity components for differential drive motion model.
+struct Velocity {
+  double v;    ///< Linear velocity (m/s)
+  double w;    ///< Angular velocity (rad/s)
+  double phi;  ///< Steering angle (rad)
+};
+
+/// Parameters to construct a AckermannDriveModel instance.
+/**
+ * See Probabilistic Robotics \cite thrun2005probabilistic Chapter 5.3, particularly table 5.3.
+ */
+struct AckermannDriveModelParam {
+  /// Rotational noise from rotational velocity
+  /**
+   * How much rotational noise is generated by the rotational velocity.
+   * Also known as `alpha1 in the differential drive model param`.
+   */
+  double rotation_noise_from_rotation;
+  /// Rotational noise from translation velocity
+  /**
+   * How much rotational noise is generated by the linear velocity.
+   * Also known as `alpha2 in the differential drive model param`.
+   */
+  double rotation_noise_from_translation;
+  /// Translational noise from translation velocity
+  /**
+   * How much translational noise is generated by the linear velocity.
+   * Also known as `alpha3 in the differential drive model param`.
+   */
+  double translation_noise_from_translation;
+  /// Translational noise from rotational velocity
+  /**
+   * How much translational noise is generated by the rotational velocity.
+   * Also known as `alpha4 in the differential drive model param`.
+   */
+  double translation_noise_from_rotation;
+  /// Additional orientation noise from translational velocity
+  /**
+   * How much extra orientation noise is generated by the linear velocity.
+   * Also known as `alpha6`.
+   */
+  double orientation_noise_from_translation;
+  /// Additional orientation noise from rotational velocity
+  /**
+   * How much extra orientation noise is generated by the rotational velocity.
+   * Also known as `alpha7`.
+   */
+  double orientation_noise_from_rotation;
+
+  /// Threshold for distinguishing between straight-line and circular motion.
+  /**
+   * Below this threshold (~0.57 degrees), motion is treated as straight-line to avoid
+   * numerical instabilities in radius calculations for nearly-zero angular velocities.
+   */
+  static constexpr double small_angle_threshold = 0.01;
+
+  /// Length of the robot (meters).
+  /**
+   * See \cite Localization and Mapping in Local Occupancy Grid Maps: Simulation
+   * in Ackerman model mobile robot by Ronald A. Cardenas , Jasper W. Huanay
+   * and Ivan Calle
+   */
```

Comment:
This is not really the lenght of the robot, but the distance between the rear and front wheel axles. See https://en.wikipedia.org/wiki/Wheelbase .

In the SingleTrack (or Bicycle) model, which is a simplification of the ackermann structure, it's the distance between the rear and front wheel.

<img width="672" height="356" alt="image" src="https://github.com/user-attachments/assets/3c3ff842-c4ab-49ae-a1bd-7a72112c643a" />